spec/facility_spec.rb

```diff
@@ -21,6 +29,208 @@
       @facility.add_service('Renew Drivers License')
       @facility.add_service('Vehicle Registration')
       expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+
+    end
+  end
+
+  describe '#registered_vehicles' do
+    it 'has registered vehicles' do
+      expect(@facility_1.registered_vehicles).to eq ([])
+    end
+  end
+
+  describe "#registration_date" do
+    it 'doesnt have a registration date' do
+      expect(@cruz.registration_date).to eq(nil)
+    end
+
+    it 'has a registration date when vehicle is registered' do
+      @facility_1.register_vehicle(@cruz)
+      expect(@cruz.registration_date).to be_an_instance_of(Date)
+    end
+  end
+
+  describe 'registration' do
+    it 'registers a vehicle' do
+
+      expect(@facility_1.register_vehicle(@cruz)).to eq ([@cruz])
+    end
+  end
+
+  describe '#collected_fees' do
+    it 'stores collected fees' do
+      expect(@facility_1.collected_fees).to eq(0)
+    end
+
+    it 'collects fees for every registered vehicle' do
+      @facility.register_vehicle(@cruz)
+      expect(@facility.registered_vehicles).to eq([@cruz])
+      @facility.fee_assignment(@cruz)
+      expect(@facility.collected_fees).to eq 100
+
+      @facility.register_vehicle(@camaro)
+      @facility.fee_assignment(@camaro)
+      expect(@facility.collected_fees).to eq 125
+    end
+
+    it 'collects all fees if meets multiple conditions' do
+      @facility.register_vehicle(@mustang)
+      @facility.fee_assignment(@mustang)
+      expect(@facility.collected_fees).to eq 225
+    end
+
+  end
+
+  describe '#plate_type' do
+    it 'has a plate type' do
+      @facility.register_vehicle(@cruz)
+      expect(@facility.registered_vehicles).to eq([@cruz])
+      @facility.plate_assignment(@cruz)
+      expect(@cruz.plate_type).to eq (:regular)
+    end
+  end
+
+  describe 'vehicle registration process' do
+    it 'collects fees and assigns plate types when registered' do
+
+      @facility_2.register_vehicle(@camaro)
+      @facility_2.register_vehicle(@bolt)
+      expect(@facility_2.registered_vehicles).to eq ([@camaro , @bolt])
+
+      @facility_2.fee_assignment(@camaro)
+      @facility_2.fee_assignment(@bolt)
+      expect(@facility_2.collected_fees).to eq(225)
+
+      @facility_2.plate_assignment(@camaro)
+      @facility_2.plate_assignment(@bolt)
+      expect(@camaro.plate_type).to eq (:antique)
+      expect(@bolt.plate_type).to eq (:ev)
+      #Need to figure out how a vehicle cannot be registered twice and/or at another facility
```

Comment:
Limiting a vehicle to one facility would be tricky and probably involve a [class method](https://backend.turing.edu/module1/lessons/class_methods). But for the other part you could add something like this to your `register_vehicle` method:
```ruby
unless @vehicles.includes?(vehicle_instance)
```